### PR TITLE
feat(blocks): add hotkey for richtext

### PR DESCRIPTION
**Pull request by Flrande**

### CODE REVIEW

**Comment by doodlewind** [comment]
When can we have test suite landed for Virgo? I would consider current stage a bit risky if we keep adding notable features without test cases.

**Comment by Flrande** [comment]
Do you mean the test for basic style of virgo?

**Comment by Flrande** [comment]
Actually, this pr has nothing to do with virgo. I add them just because Alex asked me to do that because he was unfamiliar with virgo api. What virgo will do next is support more accurate point and I will add test before develop it.

**Comment by doodlewind** [comment]
> Actually, this pr has nothing to do with virgo. I add them just because Alex asked me to do that because he was unfamiliar with virgo api. What virgo will do next is support more accurate point and I will add test before develop it.

Got it, hope to see it happen soon!

**Review by doodlewind — APPROVED** [review]

File: packages/blocks/src/database-block/components/column-type/rich-text.ts [hunk 1]
```diff
@@ -34,17 +106,82 @@ class TextCell extends DatabaseCellLitElement {
...
+    if (!this.vEditor) {
+      return;
+    }
+    const vEditor = this.vEditor;
+
+    // Bold
+    // command + b, ctrl + b
+    if (
+      (event.metaKey || event.ctrlKey) &&
+      (event.key === 'b' || event.key === 'B')
```
Comment (himself65) [inline packages/blocks/src/database-block/components/column-type/rich-text.ts]: Refactor using a jump table?

File: packages/blocks/src/database-block/components/column-type/rich-text.ts [hunk 2]
```diff
@@ -34,17 +106,82 @@ class TextCell extends DatabaseCellLitElement {
...
+      this.vEditor = new VEditor(yText, {
+        renderElement,
+        onKeyDown: this._handleKeyDown.bind(this),
+      });
       this.vEditor.mount(this._container);
       this.vEditor.focusEnd();
     }
   }
 
+  private _handleKeyDown(event: KeyboardEvent) {
```
Comment (himself65) [inline packages/blocks/src/database-block/components/column-type/rich-text.ts]:
```suggestion
  private _handleKeyDown = (event: KeyboardEvent) => {
```

File: packages/blocks/src/database-block/components/column-type/rich-text.ts [hunk 3]
```diff
@@ -34,17 +106,82 @@ class TextCell extends DatabaseCellLitElement {
         schemaId: this.column.id,
         value: yText,
       });
-      this.vEditor = new VEditor(yText);
+      this.vEditor = new VEditor(yText, {
+        renderElement,
+        onKeyDown: this._handleKeyDown.bind(this),
```
Comment (himself65) [inline packages/blocks/src/database-block/components/column-type/rich-text.ts]:
```suggestion
        onKeyDown: this._handleKeyDown,
```

File: packages/blocks/src/database-block/components/column-type/rich-text.ts [hunk 4]
```diff
@@ -34,17 +106,82 @@ class TextCell extends DatabaseCellLitElement {
...
+  }
+
   protected update(changedProperties: Map<string, unknown>) {
     super.update(changedProperties);
     if (this.tag && !this.vEditor) {
       // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      this.vEditor = new VEditor(this.tag.value as any);
+      this.vEditor = new VEditor(this.tag.value as any, {
+        renderElement,
+        onKeyDown: this._handleKeyDown.bind(this),
```
Comment (himself65) [inline packages/blocks/src/database-block/components/column-type/rich-text.ts]:
```suggestion
        onKeyDown: this._handleKeyDown,
```

**Review by Himself65 — CHANGES_REQUESTED** [review]

**Review by Himself65 — APPROVED** [review]

**Pull request merged**